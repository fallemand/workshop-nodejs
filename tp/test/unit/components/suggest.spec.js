const React = require('react');
const renderer = require('react-test-renderer');
const Suggest = require('../../../views/components/Suggest');

describe('Suggest test using snapshots', () => {
  const component = renderer.create(<Suggest query="macb" />);
  const response = {
    data: {
      results: ['macbook', 'macbook pro']
    },
  };

  afterEach(() => {
    const instance = component.getInstance();

    // Clean results.
    instance.setState({
      results: [],
    });
  });

  test('Renders empty', () => {
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  test('Renders with results (setState)', () => {
    const instance = component.getInstance();

    instance.setState(response.data)

    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  test('Renders with results (componentWillReceiveProps)', () => {
    // Mock fetch implementation.
    window.fetch = jest.fn().mockImplementation(
      () => Promise.resolve({ json: () => Promise.resolve(response) })
    );

    expect.assertions(1);

    const instance = component.getInstance();

    return instance.componentWillReceiveProps({ query: 'macb' })
      .then(() => {
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
      });
  });
});
