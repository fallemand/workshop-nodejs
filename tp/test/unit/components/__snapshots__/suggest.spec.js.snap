// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest test using snapshots Renders empty 1`] = `<div />`;

exports[`Suggest test using snapshots Renders with results (componentWillReceiveProps) 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=macbook"
      >
        macbook
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=macbook pro"
      >
        macbook pro
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest test using snapshots Renders with results (setState) 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=macbook"
      >
        macbook
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=macbook pro"
      >
        macbook pro
      </a>
    </li>
  </ul>
</div>
`;
